drafts/maui-test_pt1.md

```diff
@@ -0,0 +1,332 @@
+---
+title: How to test a .NET MAUI App Part 1
+domain: software-engineering-corner.zuehlke.com
+tags: dotnet,mvvm,xaml,net-maui,csharp,xunit,testing,moq
+cover: https://cdn.hashnode.com/res/hashnode/image/upload/v1715274155168/4q4lrfJcY.png?auto=format
+publishAs: NiZarper
+hideFromHashnodeCommunity: false
+saveAsDraft: true
+---
+
+# How to test a .NET MAUI App Part 1
+
+This blog post is the first part of a two-part series on testing a .NET MAUI application. This first post focuses on the service and view model layers, while the second will focus on the UI testing aspect.
+
+## Table Of Contents
+
+- Setup
+- Testing of a service
+- Testing of a view model
+- Takeaways
+
+## Setup
+
+This chapter provides a practical guide to the setup process for testing a .NET MAUI project.
+
+1. Add a new test project. In my case, I used XUnit.
+2. Add a reference to the .NET MAUI project you want to test.
+3. Add a UseMauiEssentials element to the project file of the test project. This will give you access to the .NET MAUI APIs inside our tests.
+4. Install wanted NuGets. In my case, I added the FluentAssertions and Moq libraries.
+
+Your project file should look similar to this:
+
+```xml
+ <Project Sdk="Microsoft.NET.Sdk">
+
+    <PropertyGroup>
+        <TargetFramework>net7.0</TargetFramework>
+        <RootNamespace>CatFinder.Tests</RootNamespace>
+        <IsPackable>false</IsPackable>
+        <IsTestProject>true</IsTestProject>
+        <UseMauiEssentials>true</UseMauiEssentials>
+    </PropertyGroup>
+
+    <ItemGroup>
+        <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.7.1" />
+        <PackageReference Include="Moq" Version="4.20.70" />
+        <PackageReference Include="xunit" Version="2.4.2" />
+        <PackageReference Include="xunit.runner.visualstudio" Version="2.4.5" />
+        <PackageReference Include="FluentAssertions" Version="6.12.0" />
+    </ItemGroup>
+
+    <ItemGroup>
+        <ProjectReference Include="..\CatFinder\CatFinder.csproj" />
+    </ItemGroup>
+
+</Project>
+```
+
+5. Add a GlobalUsing.cs. With this we can centralize the needed usings in one file.
+
+``` C#
+
+global using CatFinder.Services.CatServices;
+global using CatFinder.Services.FileSystemServices;
+
+global using CatFinder.ViewModels;
+global using CatFinder.Models;
+
+global using FluentAssertions;
+global using Xunit;
+global using Moq;
+
+global using static CatFinder.Tests.MockHelper;
+```
+
+## Testing of a service
+
+Let us now look at how we can test a service that uses the .NET MAUI filesystem.
```

Comment:
Hello Nijaz. My background is mobile developer having worked with Xamarin but not with MAUI, yet. So I was surprised to see MAUI provides it's own "filesystem". The very next thing I did was google it.
If the targeted audience includes people like me, not having worked with MAUI, yet, you might want to save them the effort by adding a short sentence of explanation.
Such as "that uses the .NET MAUI [filesystem](https://learn.microsoft.com/en-us/dotnet/maui/platform-integration/storage/file-system-helpers?view=net-maui-8.0&tabs=android), the abstraction to files that MAUI provides to  access files in the app package"

Reply:
That's a very valid point. I'm stuck between how much I want to explain to the reader, e.g., about the APIs used, and how much independence the reader brings with it. I added your suggested extension.